mscore/plugin/api/elements.h

```diff
@@ -671,6 +679,32 @@ class Measure : public Element {
       /// \endcond
       };
 
+//---------------------------------------------------------
+//   Page
+//---------------------------------------------------------
+
+class Page : public Element {
+      Q_OBJECT
+      /**
+       * Page number (read only).
+       * Returns number of this page in the score counting from 0, i.e.
+       * for the first page its \p pagenumber value will be equal to 0.
```

Comment:
Does this make sense? Why not starting with 1?

Reply:
Here I actually just made the implementation of this property equivalent to the one from MuseScore 2.X. It would be good to have a property for user-visible page numbers, but then is should start not with 1 but with `(page numbering offset setting) + 1`. Since that setting can have any value it would make it more problematic to obtain an "internal" page number, e.g. to detect the first page in a score. It seems to me it could potentially be useful to have both these kinds of page number property (or to expose a property for page numbering offset for `Score` objects) but in this pull request I just added the one which had been exposed in 2.X versions.

Reply:
Maybe it would make sense to rename it to something like `pageIndex` then?

Reply:
if it is 'just' reimplementing something from MuseScore 2.x, better keep name and semantic

Reply:
and maybe add pageOffset and let the plugin programmer sort the logic.
Was it really 'pagenumber' rather than `pageNumber'?

Reply:
Yes, it was named exactly `pagenumber`, see https://github.com/musescore/MuseScore/blob/2.3.2/libmscore/page.h#L127

I added a commit with `pageNumberOffset` property here.